internal/metrics/metrics.go

```diff
@@ -0,0 +1,174 @@
+	PDBGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eviction_autoscaler_pdbs_total",
+			Help: "Total number of PDBs seen by the eviction autoscaler",
+		},
+		[]string{"namespace", "created_by_us"},
+	)
+
+	// EvictionCounter tracks how often the eviction-autoscaler notices an eviction
+	// Labels: namespace
+	EvictionCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_evictions_total",
+			Help: "Total number of evictions noticed by the eviction autoscaler",
+		},
+		[]string{"namespace"},
+	)
+
+	// BlockedEvictionCounter tracks how often evictions are blocked by PDBs
+	// Labels: namespace, pdb_name
+	BlockedEvictionCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_blocked_evictions_total",
+			Help: "Total number of evictions blocked by PDBs",
+		},
+		[]string{"namespace", "pdb_name"},
+	)
+
+	// ScalingOpportunityCounter tracks how often the controller thinks it could have scaled a deployment
+	// Labels: namespace, deployment_name, action (scale_up/scale_down)
+	ScalingOpportunityCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_scaling_opportunities_total",
+			Help: "Total number of times the controller identified scaling opportunities",
+		},
+		[]string{"namespace", "deployment_name", "action"},
+	)
+
+	// ActualScalingCounter tracks actual scaling actions performed
+	// Labels: namespace, deployment_name, action (scale_up/scale_down)
+	ActualScalingCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_scaling_actions_total",
+			Help: "Total number of actual scaling actions performed by the controller",
+		},
+		[]string{"namespace", "deployment_name", "action"},
+	)
+
+	// PDBCreationCounter tracks PDB creation events
+	// Labels: namespace, deployment_name
+	PDBCreationCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_pdb_creations_total",
+			Help: "Total number of PDBs created by the eviction autoscaler",
+		},
+		[]string{"namespace", "deployment_name"},
+	)
+
+	// EvictionAutoScalerCreationCounter tracks EvictionAutoScaler creation events
+	// Labels: namespace, pdb_name, target_deployment
+	EvictionAutoScalerCreationCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_creation_total",
+			Help: "Total number of EvictionAutoScaler resources created",
+		},
+		[]string{"namespace", "pdb_name", "target_deployment"},
+	)
+
+	// NodeCordoningCounter tracks node cordoning events detected
+	NodeCordoningCounter = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_node_cordoning_total",
+			Help: "Total number of node cordoning events detected by the eviction autoscaler",
+		},
+	)
+
+	// PDBInfoGauge tracks various PDB-related metrics
+	// Labels: namespace, pdb_name, target_name, metric_type
+	PDBInfoGauge = prometheus.NewGaugeVec(
```

Comment:
need to talk about how this relates to pdbgauge. I don't think we want seperate gauges for each pdb we just want labels on pdb guage to tell us how many have maxunavailablle==0 and minavailable==replicas